src/sig_stfl/xmss/sig_stfl_xmss_secret_key_functions.c

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: (Apache-2.0 OR MIT) AND CC0-1.0
+
+#include <oqs/oqs.h>
+#include <string.h>
+#include <stdbool.h>
+#include "sig_stfl_xmss.h"
+
+#if defined(__GNUC__) || defined(__clang__)
+#define XMSS_UNUSED_ATT __attribute__((unused))
+#else
+#define XMSS_UNUSED_ATT
+#endif
+
+extern inline OQS_SIG_STFL_SECRET_KEY *OQS_SECRET_KEY_XMSS_new(size_t length_secret_key) {
+
+	// Initialize the secret key in the heap with adequate memory
+	OQS_SIG_STFL_SECRET_KEY *sk = malloc(sizeof(OQS_SIG_STFL_SECRET_KEY));
```

Comment:
Should we use the OQS_MEM allocs here?

Reply:
I don't know what function you refer to, the current `common.h` does not provide `OQS_MEM_malloc` or `OQS_MEM_alloc`. Can you elaborate further?

Reply:
Ah, we should probably add an `OQS_MEM_[m]alloc` for completeness, but for now nevermind. :)